code/datums/status_effects/debuffs/choke.dm

```diff
@@ -0,0 +1,300 @@
+
+/datum/status_effect/choke/proc/helped(mob/source, mob/helping)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, .proc/heimlich, source, helping)
+	return COMPONENT_BLOCK_HELP_ACT
+
+/datum/status_effect/choke/proc/shook(mob/source, mob/helping)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, .proc/heimlich, source, helping)
+	return COMPONENT_BLOCK_MISC_HELP
+
+/datum/status_effect/choke/proc/heimlich(mob/victim, mob/aggressor)
+	if(victim == aggressor)
+		return
+	if(DOING_INTERACTION_WITH_TARGET(aggressor, victim))
+		victim.balloon_alert(aggressor, "You are already helping!")
+		return
+	if(DOING_INTERACTION(aggressor, "heimlich"))
+		victim.balloon_alert(aggressor, "You can't heimlich two people at once!") // Yes I know it's kinda runon, but this is a real rare condition
+		return
+
+	if(!thrusting_continues(victim, aggressor, before_work = TRUE))
+		return
+	aggressor.start_pulling(victim)
+	// I want the thing to look vaugely like the heimlich. Sue me
+	victim.setDir(aggressor.dir)
+
+	var/hand_name = "hand"
+	if(!iscarbon(aggressor))
+		hand_name = "paw" // Fuck you
+
+	var/mob/living/livin_victim = victim
+	if(iscarbon(aggressor) && livin_victim.body_position == STANDING_UP)
+		owner.visible_message(span_warning("[aggressor] wraps [aggressor.p_their()] arms around [victim]'s stomach, and begins thrusting [aggressor.p_their()] fists towards themselves!"), \
+			span_boldwarning("[aggressor] wraps [aggressor.p_their()] arms around you, and begins thrusting their hands into your chest. [capitalize(GLOB.deity)] that hurts!"), \
+			)
+	else
+		owner.visible_message(span_warning("[aggressor] places [aggressor.p_their()] [hand_name]s on [victim]'s back, and begins forcefully striking it!"), \
+			span_boldwarning("You feel [aggressor]\s [hand_name]s on your back, and then repeated striking!"))
+
+	if(!do_after_mob(aggressor, victim, 7 SECONDS, extra_checks = CALLBACK(src, .proc/thrusting_continues, victim, aggressor), interaction_key = "heimlich"))
+		aggressor.stop_pulling()
+		return
+	aggressor.stop_pulling()
+
+	var/atom/movable/choking_on = choking_on_ref?.resolve()
+	owner.visible_message(span_green("[victim] vomits up \the[choking_on]. [victim.p_theyre()] gonna make it!"), \
+			span_green("You vomit up that accursed blockage. YOU CAN BREATH! The broken chest is a hell of a price to pay."))
+	if(iscarbon(victim))
+		var/mob/living/carbon/carbon_victim = victim
+		var/obj/item/bodypart/chest = carbon_victim.get_bodypart(BODY_ZONE_CHEST)
+		if(chest)
+			chest.force_wound_upwards(/datum/wound/blunt/severe)
+	playsound(owner, 'sound/creatures/crack_vomit.ogg', 120, extrarange = 5, falloff_exponent = 4)
+	vomit_up()
+
+/datum/status_effect/choke/proc/mirror_dir(atom/source, old_dir, new_dir)
+	SIGNAL_HANDLER
+	owner.dir = new_dir
+
+/datum/status_effect/choke/proc/thrusting_continues(mob/living/victim, mob/aggressor, before_work = FALSE)
+	if(iscarbon(aggressor))
+		var/free_hands = 0
+		// Listen bud, you need at least 2 free hands for this
```

Comment:
i became cpr certified a month ago and now i realize i hate you

Reply:
:)